src/response.rs

```diff
@@ -200,3 +220,104 @@ impl CookieResponse {
         }
     }
 }
+
+
+#[cfg(test)]
+mod tests {
+    use std::collections::BTreeMap;
+    use rustc_serialize::json::Json;
+    use super::{WebDriverResponse,
+                CloseWindowResponse,
+                CookieResponse,
+                ElementRectResponse,
+                NewSessionResponse,
+                ValueResponse,
+                TimeoutsResponse,
+                WindowPositionResponse,
+                WindowSizeResponse,
+                Cookie,
+                Nullable};
+
+    fn test(resp: WebDriverResponse, expected_str: &str) {
+        let data = resp.to_json_string();
+        let actual = Json::from_str(&*data).unwrap();
+        let expected = Json::from_str(expected_str).unwrap();
+        assert_eq!(actual, expected);
+    }
+
+    #[test]
+    fn test_close_window() {
+        let resp = WebDriverResponse::CloseWindow(
+            CloseWindowResponse::new(vec!["test".into()]));
+        let expected = r#"{"value": ["test"]}"#;
+        test(resp, expected);
+    }
+
+    #[test]
+    fn test_cookie() {
+        let resp = WebDriverResponse::Cookie(CookieResponse::new(
+            vec![
+                Cookie::new("test".into(),
+                            "test_value".into(),
+                            Nullable::Value("/".into()),
+                            Nullable::Null,
+                            Nullable::Null,
+                            true,
+                            false)
```

Comment:
This is a good example why I don’t like the `::new` constructors so much. It’s clearer what each argument is when using a struct literal:

```rust
Cookie {
    name: "test".into(),
    value: "test_value".into(),
    path: Nullable::Value("/".into()),
    …
}
```

Anyway, you don’t have to change this.